### PR TITLE
[INTEL MKL] Enable MatMul in eager mode

### PR DESCRIPTION
**Pull request by gzmkl**
Enable MKL MatMul in eager mode.

Pending on the following PRs (that is, this PR should be merged after them):
https://github.com/tensorflow/tensorflow/pull/30401   (MKL conv fwd in eager mode)
https://github.com/tensorflow/tensorflow/pull/30402   (MKL conv bwd in eager mode)

### CODE REVIEW

**Comment by agramesh1** [comment]
Pinging @penpornk for review.

**Review by penpornk — COMMENTED** [review]
This PR overlaps a lot with https://github.com/tensorflow/tensorflow/pull/30401. Can you make this PR depend on https://github.com/tensorflow/tensorflow/pull/30401? Otherwise, we should just wait until https://github.com/tensorflow/tensorflow/pull/30401 is merged.

**Comment by penpornk** [comment]
(I meant, this PR seems to have just a few lines of changes when based on https://github.com/tensorflow/tensorflow/pull/30401.)

**Comment by gzmkl** [comment]
@penpornk.  Yes, it depends on PR #30401. It is better waiting #30401 is merged. Thanks!

**Comment by gzmkl** [comment]
Headsup: after PR https://github.com/tensorflow/tensorflow/pull/30402 being merged, 
there will be some conflicts which I will resolve.

**Comment by gzmkl** [comment]
FYI - the expected merge conflicts have been resolved

File: tensorflow/core/common_runtime/eager/BUILD [hunk 1]
```diff
@@ -9,6 +9,12 @@ load(
     "mkl_deps",
 )
 
+load(
```
Comment (penpornk) [inline tensorflow/core/common_runtime/eager/BUILD]: This has already been loaded in the lines above. Please remove.

File: tensorflow/core/common_runtime/eager/BUILD [hunk 2]
```diff
@@ -272,6 +278,7 @@ cc_library(
             "//tensorflow/core/distributed_runtime/eager:remote_copy_node",
         ],
     }) + if_mkl([":mkl_eager_op_rewrite"]),
+
```
Comment (penpornk) [inline tensorflow/core/common_runtime/eager/BUILD]: Please remove the extra blank line.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you! A few minor changes please.

**Comment by agramesh1** [comment]
Please ignore, Git merge messed things up. Created a new PR here https://github.com/tensorflow/tensorflow/pull/31311  I will try to close this PR.

**Comment by penpornk** [comment]
I'll close the PR for you.

**Pull request closed without merging**